source/common/stats/metric_impl.h

```diff
@@ -26,13 +26,17 @@ class MetricImpl : public virtual Metric {
              SymbolTable& symbol_table);
   ~MetricImpl();
 
-  std::string name() const override { return symbolTable().toString(statName()); }
+  std::string name() const override { return constSymbolTable().toString(statName()); }
   std::string tagExtractedName() const override;
   std::vector<Tag> tags() const override;
   StatName tagExtractedStatName() const override;
   void iterateTagStatNames(const TagStatNameIterFn& fn) const override;
   void iterateTags(const TagIterFn& fn) const override;
 
+  const SymbolTable& constSymbolTable() const override {
+    return const_cast<MetricImpl*>(this)->symbolTable();
```

Comment:
Is this const_cast needed? Shouldn't it automatically be convertable to a const object?

Reply:
The const-cast is to take the const MetricImpl (this) and turn it into a non-const one, so we can use the non-const helper method, which must be defined by a sub-class.  I think I'll go ahead and add that as a comment.

BTW I find this is a common pattern when defining two functionally equivalent but non-trivial accessors, one of which is a const method returning a const ref/ptr, the other is a non-const method returning a non-const ref/ptr. If you want to have a single implementation you need at least one const-cast.

I could also have had subclasses define the const method, in which case I'd need to const-cast the returned object (SymbolTable&) rather than const-casting this. I have a slightly preference for doing it this way but don't have a particularly strong reason for that.